### DISCUSSION TITLE
Migtate mysql external database

### DISCUSSION THREAD

**Discussion by katsil** [k3s-io/k3s#8869] [Q&A]
Hi. Currently i am running k3s cluster with 10 servers and set up it with only masters. It is okay in my setup.

The bottle neck is a mysql vps server and i tried to migrate to a xtradb cluster but the perfomanse is very slow  (if you can suggest some thoughts why this is happening or how to tune xtradb i would be very thankful). But anyway is there any way to migrate mysql external datastore to a etcd datastore for example?

**Comment by brandond** [comment]
> i am running k3s cluster with 10 servers

Have you considered just rebuilding this so that only 2 of your nodes are servers, and the rest are agents? There's not a ton of value, but a lot of overhead, in having all the nodes be servers.

> is there any way to migrate mysql external datastore to a etcd datastore

No, there is no supported way to do this.

**Reply by katsil** [reply]
> There's not a ton of value, but a lot of overhead

in overhead i just see a lot of `select` commands in mysql and a lot of connections. 

So can you suggest and help how to move from k3s master server to 'worker'/'node' in production cluster? Or there is only one way draining/removing node and re-initializating?

**Reply by brandond** [reply]
Technically you can just run `k3s agent` instead of running `k3s server` on these nodes to make the switch, but you'd have to change the systemd units from `k3s` to `k3s-agent` and modify the configuration by hand. It'd probably be easier to one-by-one drain the nodes and uninstall/reinstall k3s in your desired configuration.
